lib/galaxy/model/metadata.py

```diff
@@ -604,7 +604,16 @@ def wrap(self, value, session):
         if isinstance(value, int):
             return session.query(galaxy.model.MetadataFile).get(value)
         else:
-            return session.query(galaxy.model.MetadataFile).filter_by(uuid=value).one()
+            wrapped_value = session.query(galaxy.model.MetadataFile).filter_by(uuid=value).one_or_none()
```

Comment:
Maybe factor out into a function? Then the logic is so much easier to read, I think:
```py
wrapped_value = get_wrapped_value(session, value)
if wrapped_value:
    return wrapped_value
else:
    # commit here
    return get_wrapped_value(session, value)
...
def get_wrapped_value(session, value):
    stmt = select(galaxy.model.MetadataFile).filter_by(uuid=value)
    return session.scalars(stmt).one_or_none()
```

Reply:
This is already conflicting (https://github.com/galaxyproject/galaxy/pull/16933) bugfix on a released branch on notoriously problematic module. I think refactoring should go against dev afterwards. That said, I am sorry I did not communicate this clearly before merging.

I will open that refactoring PR once we resolve the conflict.

Reply:
No problem, I can fix this in a follow-up. It was not refactoring - it was new code added (minus sqlalchemy 2.0 syntax), which is why I suggested the change. Again, that's OK. I'll take a look at the conflict.